test/forwarder/forwarder.js

```diff
@@ -105,61 +115,247 @@ contract('Forwarder', async (accounts) => {
     });
 
-    it ('In case of invalid request(data) is not different from signature, it should not be verified in the forwarder contract', async () => {
+    it ('In case of invalid request(data), it should not be verified and should be reverted in executing of the forwarder contract', async () => {
         const request_other =  {
             from: relayer1Address,
-            to: BridgeAddress,
+            to: TestTargetInstance.address,
             value: '0',
             gas: '300000',
             nonce: 0,
             data: '0x1234'
         }
 
         assert.equal((await ForwarderInstance.verify(request_other, sign)), false);
+        return TruffleAssert.reverts(ForwarderInstance.execute(request_other, sign), "MinimalForwarder: signature does not match request");
+    });
+
+    it ('If signature is valid, but req.from != signer, it should be reverted and should not be verified', async () => {
+        const request_other =  {
+            from: relayer2Address,
+            to: TestTargetInstance.address,
+            value: '0',
+            gas: '300000',
+            nonce: 0,
+            data: '0x'
+        }
+
+        assert.equal((await ForwarderInstance.verify(request_other, sign)), false);
+        return TruffleAssert.reverts(ForwarderInstance.execute(request_other, sign), "MinimalForwarder: signature does not match request");
+    });
+
+    it ('If signature is valid, but req.nonce != nonce[signer], it should be reverted and should not be verified', async () => {
+        const request_other =  {
+            from: relayer1Address,
+            to: TestTargetInstance.address,
+            value: '0',
+            gas: '300000',
+            nonce: 10,
+            data: '0x'
+        }
+
+        assert.equal((await ForwarderInstance.verify(request_other, sign)), false);
+        return TruffleAssert.reverts(ForwarderInstance.execute(request_other, sign), "MinimalForwarder: signature does not match request");
+    });
+
+    it ('Execute should succeed even if the call to the target failed', async () => {
+        const new_domain = {
+            name,
+            version,
+            chainId: 1,
+            verifyingContract: ResponseForwarderInstance.address,
+        };
+
+        const new_request = {
+            from: relayer1Address,
+            to: ForwarderInstance.address,
+            value: '0',
+            gas: '300000',
+            nonce: 0,
+            data: '0x'
+        }
+
+        const new_sign = ethSigUtil.signTypedMessage(
+            relayer1.getPrivateKey(),
+            {
+                data: {
+                    types: types,
+                    domain: new_domain,
+                    primaryType: 'ForwardRequest',
+                    message: new_request
+                }
+            }
+        )
+
+        await ResponseForwarderInstance.response(new_request, new_sign);
+        const status = await ResponseForwarderInstance.status();
```

Comment:
Could use a normal forwarder and do `await ForwarderInstance.execute.call(new_request, new_sign)` to get the output.
Current way also works, but the problem is that it tests a modified version of the Forwarder which could behave differently.